api/fixtures/example.go

```diff
@@ -656,11 +658,25 @@ func (o ExampleOptions) Resources() *ExampleResources {
 			}
 			nodePool.Annotations[hyperv1.AllowUnsupportedKubeVirtRHCOSVariantsAnnotation] = val
 		}
-	case hyperv1.NonePlatform, hyperv1.AgentPlatform:
+	case hyperv1.NonePlatform:
+		nodePool := defaultNodePool(cluster.Name)
+		if nodePool.Spec.Management.UpgradeType == "" {
+			nodePool.Spec.Management.UpgradeType = hyperv1.UpgradeTypeInPlace
+		}
+		nodePools = append(nodePools, nodePool)
+	case hyperv1.AgentPlatform:
 		nodePool := defaultNodePool(cluster.Name)
+		nodePool.Spec.Platform.Agent = &hyperv1.AgentNodePoolPlatform{}
 		if nodePool.Spec.Management.UpgradeType == "" {
 			nodePool.Spec.Management.UpgradeType = hyperv1.UpgradeTypeInPlace
 		}
+		if o.Agent.AgentLabelSelector != "" {
+			agentSelector := metav1.LabelSelector{}
+			if err := json.Unmarshal([]byte(o.Agent.AgentLabelSelector), &agentSelector); err != nil {
```

Comment:
Is there a reason we don't parse the label selector from a string as in other kube cli commands? 
It seems unintuitive to expect a valid json doc to be specified.
https://github.com/kubernetes/apimachinery/blob/46a872ec555cd60a5bc172ce28d9ae0234266aae/pkg/labels/selector.go#L843-L874

Reply:
Great suggestion, thanks @csrwng!  That method specifically returns the wrong kind of selector, but I used a similar one.